File: solutions/007.reverse-integer/reverse-integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        new_x = 0
        sign = 1 if x > 0 else -1
        x = x if x > 0 else 0 - x
        while x:
            tmp = x % 10
            x = x / 10
            new_x = 10 * new_x + tmp
            
        return 0 if new_x > math.pow(2, 31) else new_x * sign
       
        

File: solutions/007.reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
        long rev = 0;
        while (x != 0) {
            int tmp = x % 10;
            rev = rev * 10 + tmp;
            x = x /10;
        }
        
        if (rev > Integer.MAX_VALUE || rev < Integer.MIN_VALUE){
            rev = 0L;
        }
        
        return (int)rev;
    }
}